src/System.Linq/tests/ExceptTests.cs

```diff
@@ -0,0 +1,196 @@
+            string[] second = { null };
+            string[] expected = { string.Empty };
+
+            Assert.Equal(expected, first.Except(second, EqualityComparer<string>.Default));
+        }
+
+        [Fact]
+        public void DoubleNullWithEmpty()
+        {
+            string[] first = { null, null };
+            string[] second = new string[0];
+            string[] expected = { null };
+
+            Assert.Equal(expected, first.Except(second, EqualityComparer<string>.Default));
+        }
+
+        [Fact]
+        public void EmptyWithNonEmpty()
+        {
+            int[] first = { };
+            int[] second = { -6, -8, -6, 2, 0, 0, 5, 6 };
+            int[] expected = { };
+
+            Assert.Equal(expected, first.Except(second));
+        }
+
+        [Fact]
+        public void NonEmptyWithEmpty()
+        {
+            int?[] first = { -6, -8, -6, 2, 0, 0, 5, 6, null, null };
+            int?[] second = { };
+            int?[] expected = { -6, -8, 2, 0, 5, 6, null };
+
+            Assert.Equal(expected, first.Except(second));
+        }
+
+        [Fact]
+        public void EachHasRepeatsBetweenAndAmonthThemselves()
+        {
+            int?[] first = { 1, 2, 2, 3, 4, 5 };
+            int?[] second = { 5, 3, 2, 6, 6, 3, 1, null, null };
+            int?[] expected = { 4 };
+
+            Assert.Equal(expected, first.Except(second));
+        }
+
+        [Fact]
+        public void ReatedInFirstAndAllInSecondNotInFirst()
+        {
+            int[] first = { 1, 1, 1, 1, 1 };
+            int[] second = { 2, 3, 4 };
+            int[] expected = { 1 };
+
+            Assert.Equal(expected, first.Except(second));
+        }
+
+        [Fact]
+        public void OrderPreserved()
+        {
+            int?[] first = { 2, 3, null, 2, null, 4, 5 };
+            int?[] second = { 1, 9, null, 4 };
+            int?[] expected = { 2, 3, 5 };
+
+            Assert.Equal(expected, first.Except(second));
+        }
+
+        [Fact]
+        public void NullEqualityComparer()
+        {
+            string[] first = { "Bob", "Tim", "Robert", "Chris" };
+            string[] second = { "bBo", "shriC" };
+            string[] expected = { "Bob", "Tim", "Robert", "Chris" };
+
+            Assert.Equal(expected, first.Except(second, null));
+        }
+
+        [Fact]
+        public void CustomComparer()
+        {
+            string[] first = { "Bob", "Tim", "Robert", "Chris" };
+            string[] second = { "bBo", "shriC" };
+            string[] expected = { "Tim", "Robert" };
+
+            Assert.Equal(expected, first.Except(second, new AnagramEqualityComparer()));
+        }
+
+        [Fact]
+        public void FirstNullCustomComparer()
+        {
+            string[] first = null;
+            string[] second = { "bBo", "shriC" };
+
+            var ane = Assert.Throws<ArgumentNullException>(() => first.Except(second, new AnagramEqualityComparer()));
+            Assert.Equal("first", ane.ParamName);
```

Comment:
I believe this could be done slightly more concisely as:
```C#
Assert.Throws<ArgumentNullException>("first", () => first.Except(second, new AnagramEqualityComparer()));
```
but it's not a big deal either way.